graph/src/blockchain/polling_block_stream.rs

```diff
@@ -487,23 +485,15 @@ where
         Ok(ptr)
     }
 
-    /// Set subgraph deployment entity synced flag if and only if the subgraph block pointer is
-    /// caught up to the head block pointer.
-    fn update_subgraph_synced_status(&self) -> Result<(), StoreError> {
-        let head_ptr_opt = self.chain_store.chain_head_ptr()?;
-        let subgraph_ptr = self.current_block.clone();
-
-        if head_ptr_opt != subgraph_ptr || head_ptr_opt.is_none() || subgraph_ptr.is_none() {
-            // Not synced yet
-            Ok(())
-        } else {
-            // Synced
-
+    /// Stop metrics if subgraph is synced.
+    /// Sync mechanism is now implemented in the instance manager.
+    async fn handle_subgraph_synced(&self) -> Result<(), StoreError> {
+        if self.subgraph_store.is_deployment_synced().await? {
             // Stop recording time-to-sync metrics.
             self.metrics.stopwatch.disable();
```

Comment:
This should be move to the instance manager, which should allow us to remove the dependency on the `subgraph_store` entirely.

Reply:
I'll refactor this!

Reply:
Fixed! Also I've removed the unnecessary `+ Send` trait bound of `Blockchain::TriggerData` since it's not needed anymore.

If you think I should keep it, please inform me 😊

Reply:
I removed the dependency completely!!!